README.md

```diff
@@ -1,23 +1,54 @@
-The AuthN/Z providers are defined in `package.json` and `config.ts`. You can choose  appropriate providers. SMART allows greater granularity into authentication than Cognito and is the FHIR standard. It allows you to access a FHIR record only if that record has reference to the user.
-
+FHIR Works on AWS is a software toolkit that can be used to deploy a [FHIR server](https://www.hl7.org/fhir/overview.html) on AWS. Using this toolkit, you can customize and add different FHIR functionality to best serve your use cases.
 
 ## FHIR Works on AWS features
 
-FHIR Works on AWS utilizes AWS Lambda, Amazon DynamoDB, Amazon S3 and Amazon Elasticsearch Service to provide the following FHIR features:
+FHIR Works on AWS implements the FHIR specification (STU3 or R4) and deploys that implementation into an AWS account using a set of mostly serverless AWS services.  It provides the following FHIR features:
+
++ Create, Read, Update, Delete (CRUD) operations for *all* R4 or STU3 base FHIR resources
++ Search by FHIR resource type
++ Versioned reads ([vread](https://www.hl7.org/fhir/http.html#vread))
++ Transaction bundle POSTs of up to 25 entries
++ Third-party Implementation Guides for localization
++ Authentication and authorization using either Role-Based Access Control or SMART
+
+## Architecture
+
+The system architecture comprises Amazon API Gateway, AWS Lambda, Amazon DynamoDB, Amazon S3, Amazon Elasticsearch Service, and optionally Amazon Cognito, plus CloudWatch Logs. The endpoint is published through API Gateway. The persistence layer (database and storage) uses Amazon DynamoDB and S3, with Elasticsearch as the search index for the data initially written to DynamoDB. The endpoint is secured by API keys, and by Cognito for user-level authentication and user-group authorization. CloudWatch Logs capture system events.  The diagram below shows the toolkit's architecture components and how they are related.
+ 
+![Architecture](resources/architecture.png)
+
+## Authentication and Authorization options
+
+[Cognito and Role-Based Access Control](https://github.com/awslabs/fhir-works-on-aws-authz-rbac) is deployed by default, since it includes an Identity Provider (Cognito User Pools) and is easy to configure. However, its permissions model is less granular than is needed for some use cases: when a new user is created, it is assigned to the auditor, practitioner, or non-practitioner groups. The FHIR resource access granted to the user depends on their group membership.  
+
+If preferred, [SMART](https://github.com/awslabs/fhir-works-on-aws-authz-smart) can be used instead for authentication and authorization to the FHIR server’s resources. It offers more granularity for authentication, permitting access to a particular resource only if it has a reference to the requesting user identity, and is an increasingly widely-used supplementary standard to FHIR itself. It does require being configured to use a third-party OpenID Connect Identity Provider, though.  
```

Comment:
Suggested: Please omit, "If preferred". Else, we can also say "SMART can also be used for authentication..."

Suggest we write the term as "Open ID Connect identity provider". Ref: https://openid.net/connect/

Please omit ", though" in the end.